app/javascript/react/screens/App/Overview/screens/PlanWizard/PlanWizard.js

```diff
@@ -5,25 +5,88 @@ import { createMigrationPlans } from './helpers';
 import PlanWizardBody from './PlanWizardBody';
 import { MIGRATIONS_FILTERS } from '../../OverviewConstants';
 
-const planWizardSteps = [
-  'planWizardGeneralStep',
-  'planWizardVMStep',
-  'planWizardAdvancedOptionsStep',
-  'planWizardScheduleStep',
-  'planWizardResultsStep'
-];
+import componentRegistry from '../../../../../../components/componentRegistry';
+import PlanWizardGeneralStep from '../PlanWizard/components/PlanWizardGeneralStep';
+import PlanWizardScheduleStep from '../PlanWizard/components/PlanWizardScheduleStep';
 
 class PlanWizard extends React.Component {
+  planWizardVMStepContainer = componentRegistry.markup('PlanWizardVMStepContainer');
+  planWizardResultsStepContainer = componentRegistry.markup('PlanWizardResultsStepContainer');
+  planWizardAdvancedOptionsStepContainer = componentRegistry.markup('PlanWizardAdvancedOptionsStepContainer');
+  planWizardInstancePropertiesStepContainer = componentRegistry.markup('PlanWizardInstancePropertiesStepContainer');
+
   state = { activeStepIndex: 0 };
 
+  getWizardSteps = () => {
+    const { planWizardGeneralStep, transformationMappings } = this.props;
+
+    const steps = [
+      {
+        id: 'planWizardGeneralStep',
+        title: __('General'),
+        render: () => <PlanWizardGeneralStep />,
+        disableGoto: !this.props.planWizardGeneralStep.values
+      },
+      {
+        id: 'planWizardVMStep',
+        title: __('VMs'),
+        render: () => this.planWizardVMStepContainer,
+        disableGoto: !this.props.planWizardVMStep.values
+      },
+      {
+        openstackSpecific: true,
```

Comment:
I'm concerned about the scalability of this approach.  While there are no definite plans to add other target providers, I think it would be wise to build in some flexibility to do so.

Perhaps it would be better to insert a step, [instead of filtering](https://github.com/ManageIQ/miq_v2v_ui_plugin/pull/547/files#diff-a7507d1fbaba12698eb13826effa926bR72)

Reply:
I was originally thinking we could just iterate on that filtering behavior, but I think you're right that it'll get confusing with more target providers. I'll rework this to be a little more flexible.